packages/utils/src/feature-flags/utils/flag-router.ts

```diff
@@ -1,17 +1,59 @@
 import { FeatureFlagTypes } from "@medusajs/types"
 
 export class FlagRouter implements FeatureFlagTypes.IFlagRouter {
-  private readonly flags: Record<string, boolean> = {}
+  private flags: Record<string, boolean | Record<string, boolean>> = {}
 
-  constructor(flags: Record<string, boolean>) {
+  constructor(flags: Record<string, boolean | Record<string, boolean>>) {
     this.flags = flags
   }
 
-  public isFeatureEnabled(key: string): boolean {
-    return !!this.flags[key]
+  /**
+   * Check if a feature flag is enabled.
+   * There are two ways of using this method:
+   * 1. `isFeatureEnabled("myFeatureFlag")`
+   * 2. `isFeatureEnabled({ myNestedFeatureFlag: "someNestedFlag" })`
+   * We use 1. for top-level feature flags and 2. for nested feature flags. Almost all flags are top-level.
+   * An example of a nested flag is workflows. To use it, you would do:
+   * `isFeatureEnabled({ workflows: Workflows.CreateCart })`
+   * @param flag - The flag to check
+   * @return {boolean} - Whether the flag is enabled or not
+   */
+  public isFeatureEnabled(flag: string | Record<string, string>): boolean {
+    if (typeof flag === `string`) {
+      return !!this.flags[flag]
+    }
+
+    const [nestedFlag, value] = Object.entries(flag)[0]
+
+    return !!this.flags[nestedFlag]?.[value]
```

Comment:
thought: should we also allow `{ workflows: true }` for blanket enabling workflows?

Reply:
That's a great idea - will add!

Reply:
This is actually already possible, because this `return !!this.flags[flag]` returns the truthy value.

Reply:
Maybe I missed something, but that seems true only if the flag attribute is a string. Currently we're passing objects for workflows granularly.

`isFeatureEnabled({ workflows: Workflows.CreateProducts })` should return true if the project config is `{ workflows: true }`

Reply:
Ahh sorry - I misunderstood what you meant. This is also a good addition 😄